src/geo/map.js

```diff
@@ -21,24 +21,78 @@ var Map = Model.extend({
     provider: 'leaflet'
   },
 
-  initialize: function() {
+  initialize: function (attrs, options) {
+    options = options || {};
     this.layers = new Layers();
+    this.geometries = new Backbone.Collection();
+
+    this._windshaftMap = options.windshaftMap;
+    this._dataviewsCollection = options._dataviewsCollection;
+
+    this._initBinds();
+  },
 
-    this.layers.bind('reset', function() {
-      if(this.layers.size() >= 1) {
+  _initBinds: function () {
+    this.layers.bind('reset', function () {
+      if (this.layers.size() >= 1) {
         this._adjustZoomtoLayer(this.layers.models[0]);
       }
     }, this);
 
-    this.layers.bind('reset', this._updateAttributions, this);
-    this.layers.bind('add', this._updateAttributions, this);
-    this.layers.bind('remove', this._updateAttributions, this);
+    // TODO: When the order of the layers change, the instance of the map needs
+    // to be re-recreated
+    this.layers.bind('reset', this._onLayersResetted, this);
+    this.layers.bind('add', this._onLayerAdded, this);
+    this.layers.bind('remove', this._onLayerRemoved, this);
     this.layers.bind('change:attribution', this._updateAttributions, this);
 
-    this.geometries = new Backbone.Collection();
+    if (this._dataviewsCollection) {
+
+      // When new dataviews are defined, a new instance of the map needs to be created
+      this._dataviewsCollection.bind('add', _.debounce(this._onDataviewAdded, 10), this);
+    }
+  },
+
+  _onLayersResetted: function () {
+    this.reload();
+    this._updateAttributions();
+  },
+
+  _onLayerAdded: function (layerModel) {
+    this.reload({
+      sourceLayerId: layerModel.get('id')
+    });
+    this._updateAttributions();
+  },
+
+  _onLayerRemoved: function () {
+    this.reload();
+    this._updateAttributions();
+  },
+
+  _onDataviewAdded: function (layerModel) {
+    this.reload();
   },
 
-  _updateAttributions: function() {
+  reload: function (options) {
+    this._createWindshaftMapInstance(options);
```

Comment:
If this is the only usage of this method, why not just inline the content of that function here instead?